### PR TITLE
[Perf] Use Java Version Tooling to Configure Version Changes in Perf Tooling

### PR DESCRIPTION
**Pull request by alzimmermsft**
Updates the Java Perf framework to use the standard version tooling found in `azure-sdk-for-java`.

### CODE REVIEW

**Comment by alzimmermsft** [comment]
Example of this in action: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1821644&view=results

**Review by mikeharder — CHANGES_REQUESTED** [review]
I'm just starting to look at this, but I'm concerned this might have broken the "package versions" run.  Before this PR, for the "package versions" run we saw:

```
[07:10:16.943] [INFO] Reactor Build Order:
[07:10:16.943] [INFO] 
[07:10:16.944] [INFO] Microsoft Azure SDK for Java Parent                                [pom]
[07:10:16.944] [INFO] Maven Build Tools                                                  [jar]
[07:10:16.944] [INFO] Microsoft Azure SDK for Java - Client Libraries                    [pom]
[07:10:16.944] [INFO] Microsoft Azure SDK for Java - Performance Test Libraries Parent   [pom]
[07:10:16.944] [INFO] Microsoft Azure Java Performance Test Library                      [jar]
[07:10:16.945] [INFO] azure-storage-perf                                                 [jar]
```

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1823347&view=logs&j=dc42670d-1473-57cd-faea-36a5bc05eab2&t=dfb494f3-2da1-55d1-b8fd-9d227ffe033a&l=292

After this PR:

```
[01:28:25.502] [INFO] Reactor Build Order:
[01:28:25.502] [INFO] 
[01:28:25.502] [INFO] Microsoft Azure SDK for Java Parent                                [pom]
[01:28:25.502] [INFO] Maven Build Tools                                                  [jar]
[01:28:25.502] [INFO] Microsoft Azure SDK for Java - Client Libraries                    [pom]
[01:28:25.502] [INFO] Microsoft Azure Java Core Library                                  [jar]
[01:28:25.502] [INFO] Microsoft Azure Java Core Test Library                             [jar]
[01:28:25.502] [INFO] Microsoft Azure Netty HTTP Client Library                          [jar]
[01:28:25.502] [INFO] Microsoft Azure OkHttp HTTP Client Library                         [jar]
[01:28:25.502] [INFO] Microsoft Azure SDK for Java - Performance Test Libraries Parent   [pom]
[01:28:25.502] [INFO] Microsoft Azure Java Performance Test Library                      [jar]
[01:28:25.502] [INFO] Microsoft Azure common module for Storage                          [jar]
[01:28:25.502] [INFO] Microsoft Azure internal Avro module for Storage                   [jar]
[01:28:25.502] [INFO] Microsoft Azure client library for Blob Storage                    [jar]
[01:28:25.502] [INFO] Microsoft Azure client library for Blob Storage cryptography       [jar]
[01:28:25.502] [INFO] Microsoft Azure client library for File Share Storage              [jar]
[01:28:25.502] [INFO] Microsoft Azure client library for File Storage Data Lake          [jar]
[01:28:25.502] [INFO] azure-storage-perf                                                 [jar]
```

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1825915&view=logs&j=dc42670d-1473-57cd-faea-36a5bc05eab2&t=dfb494f3-2da1-55d1-b8fd-9d227ffe033a&l=81

I will move https://github.com/Azure/azure-sdk-tools/pull/3716 (Print runtime package versions) to the top of my backlog, and see if I can get this implemented for Java and merged tomorrow.  Then we can verify the Java runtime package versions before/after this PR more easily, without relying on the build logs.

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Java.cs [hunk 1]
```diff
@@ -198,5 +203,21 @@ public override Task CleanupAsync(string project)
...
         }
+
+        private IDictionary<string, string> LoadSourceVersions()
+        {
+            var sourceVersions = new Dictionary<string, string>();
+            foreach (var line in File.ReadAllLines(VersionFile))
+            {
+                var trimmedLine = line.Trim();
+                if (string.IsNullOrEmpty(trimmedLine)) continue;
+                if (trimmedLine.StartsWith('#') || trimmedLine.StartsWith("beta_") || trimmedLine.StartsWith("unreleased_")) continue;
```
Comment (mikeharder) [inline tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Java.cs:214]:
```suggestion
                if (string.IsNullOrEmpty(trimmedLine) ||
                    trimmedLine.StartsWith('#') ||
                    trimmedLine.StartsWith("beta_", StringComparison.Ordinal) ||
                    trimmedLine.StartsWith("unreleased_", StringComparison.Ordinal))
                {
                    continue;
                }
```

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Java.cs [hunk 2]
```diff
@@ -67,20 +72,20 @@ private static void UpdatePackageVersions(string projectFile, IDictionary<string
                 var nameParts = v.Key.Split(':');
                 var groupId = nameParts[0];
                 var artifactId = nameParts[1];
+                var packageVersion = v.Value;
+
+                if (packageVersion == Program.PackageVersionSource)
+                {
+                    if (!sourceVersions.TryGetValue(v.Key, out packageVersion)) continue;
```
Comment (mikeharder) [inline tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Java.cs:79]:
```suggestion
                    if (!sourceVersions.TryGetValue(v.Key, out packageVersion))
                    {
                        continue;
                    }
```

**Review by mikeharder — APPROVED** [review]
Approved pending formatting changes

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Java.cs (hunk 1 above)
Comment (mikeharder) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Java.cs:214]: Or you could use `StringComparison.OrdinalIgnoreCase` if you want to match any case.

**Pull request merged**